src/pybind/mgr/influx/module.py

```diff
@@ -0,0 +1,211 @@
+import json
+import sys
+import time
+from ConfigParser import SafeConfigParser
+from collections import defaultdict
+from influxdb import InfluxDBClient
+from mgr_module import MgrModule
+from datetime import datetime
+from threading import Event 
+
+class Module(MgrModule):
+ 
+    run = True 
+    COMMANDS = [
+        {
+            "cmd": "influx debug",
+            "desc": "debug the module",
+            "perm": "rw"  
+        },
+    ]
+
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self.event = Event()
+
+
+    def get_latest(self, daemon_type, daemon_name, stat):
+        data = self.get_counter(daemon_type, daemon_name, stat)[stat]
+        if data:
+            return data[-1][1]
+        else:
+            return 0
+
+
+    def get_rate(self, daemon_type, daemon_name, stat):
+        data = self.get_counter(daemon_type, daemon_name, stat)[stat]
+        if data and len(data) > 1:
+            return (data[-1][1] - data[-2][1]) / float(data[-1][0] - data[-2][0])
+        else:
+            return 0
+
+
+    def sum_stat(self, type_inst):
+        path = "osd." + type_inst.__str__()
+        osdmap = self.get("osd_map")
+        value = 0
+        for osd in osdmap['osds']:
+            osd_id = osd['osd']
+            value += self.get_latest("osd", osd_id.__str__(), path.__str__())
+        return int(value) 
+     
+
+    def get_stat(self, type_inst):
+        path = "osd." + type_inst.__str__()
+        osdmap = self.get("osd_map")
+        data = []
+        for osd in osdmap['osds']:
+            osd_id = osd['osd']
+            metadata = self.get_metadata('osd', "%s" % osd_id)
+            point = {
+                "measurement": "ceph_osd_stats",
+                "tags": {
+                    "mgr_id": self.get_mgr_id(),
+                    "osd_id": osd_id,
+                    "type_instance": type_inst,
+                    "host": metadata['hostname']
+                },
+                    "time" : datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ'), 
```

Comment:
You can avoid the need for a format string here by just doing .isoformat() + "Z"